### PR TITLE
Set path to save pseudo masks into workspace

### PR DESCRIPTION
**Pull request by sungchul1**
### Summary

This PR is to set a path where saving pseudo masks, which are used on self-supervised learning for semantic segmentation, into workspace.

### How to test

When workspace was set to `test4/workspace`, pseudo masks were created in the path.
image

tox -e tests-all-py38 -- tests/unit/core/data/adapter/test_segmentation_adapter.py::TestSelfSLSegmentationDatasetAdapter

image

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by jaegukhyun** [comment]
Please, rebase to release1.3.0

**Comment by sungchul1** [comment]
> Please, rebase to release1.3.0

Fixed.

**Comment by jaegukhyun** [comment]
Could you check pre-commit issues?

File: otx/cli/manager/config_manager.py [hunk 1]
```diff
@@ -406,6 +406,11 @@ def get_dataset_config(self, subsets: List[str], hyper_parameters: Optional[Conf
             if learning_parameters:
                 num_workers = getattr(learning_parameters, "num_workers", 0)
                 dataset_config["cache_config"]["num_workers"] = num_workers
+        if str(self.task_type == "SEGMENTATION").upper() and str(self.train_type).upper() == "SELFSUPERVISED":
```
Comment (yunchu) [inline otx/cli/manager/config_manager.py]:
```suggestion
        if str(self.task_type).upper() == "SEGMENTATION" and str(self.train_type).upper() == "SELFSUPERVISED":
```
do you mean something like this?

Comment (sungchul1) [reply otx/cli/manager/config_manager.py]: Oh, thanks for finding my mistake!

File: otx/core/data/adapter/segmentation_dataset_adapter.py [hunk 2]
```diff
@@ -199,23 +204,21 @@ def _import_dataset(
         self.is_train_phase = True
 
         # Load pseudo masks
-        img_dir = None
         total_labels = []
+        print(pseudo_mask_dir)
```
Comment (sungchul1) [inline otx/core/data/adapter/segmentation_dataset_adapter.py]:
```suggestion

```
@sungmanc we don't need this, right?

**Review by jaegukhyun — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**